test/test_container_token_manager.py

```diff
@@ -0,0 +1,275 @@
+        response = responses.CallbackResponse(
+            method=responses.POST,
+            url='https://iam.cloud.ibm.com/identity/token',
+            callback=callback,
+        )
+
+        responses.add(response)
+
+        func()
+
+    return wrapper
+
+
+@mock_iam_response
+def test_request_token_auth_default():
+    iam_url = "https://iam.cloud.ibm.com/identity/token"
+
+    token_manager = ContainerTokenManager(
+        cr_token_filename=cr_token_file,
+        iam_profile_name=MOCK_IAM_PROFILE_NAME,
+    )
+    token_manager.request_token()
+
+    assert len(responses.calls) == 1
+    assert responses.calls[0].request.url == iam_url
+    assert responses.calls[0].request.headers.get('Authorization') is None
+    assert json.loads(responses.calls[0].response.text)['access_token'] == TEST_ACCESS_TOKEN_1
+
+
+@mock_iam_response
+def test_request_token_auth_in_ctor():
+    default_auth_header = 'Basic Yng6Yng='
+    token_manager = ContainerTokenManager(
+        cr_token_filename=cr_token_file,
+        iam_profile_name=MOCK_IAM_PROFILE_NAME,
+        client_id='foo',
+        client_secret='bar')
+
+    token_manager.request_token()
+
+    assert len(responses.calls) == 1
+    assert responses.calls[0].request.headers['Authorization'] != default_auth_header
+    assert json.loads(responses.calls[0].response.text)['access_token'] == TEST_ACCESS_TOKEN_1
+    assert 'scope' not in responses.calls[0].response.request.body
+
+
+@mock_iam_response
+def test_request_token_auth_in_ctor_with_scope():
+    default_auth_header = 'Basic Yng6Yng='
+    token_manager = ContainerTokenManager(
+        cr_token_filename=cr_token_file,
+        iam_profile_name=MOCK_IAM_PROFILE_NAME,
+        client_id='foo',
+        client_secret='bar',
+        scope='john snow')
+
+    token_manager.request_token()
+
+    assert len(responses.calls) == 1
+    assert responses.calls[0].request.headers['Authorization'] != default_auth_header
+    assert json.loads(responses.calls[0].response.text)['access_token'] == TEST_ACCESS_TOKEN_1
+    assert 'scope=john+snow' in responses.calls[0].response.request.body
+
+
+def test_retrieve_cr_token_success():
+    token_manager = ContainerTokenManager(
+        cr_token_filename=cr_token_file,
+    )
+
+    cr_token = token_manager.retrieve_cr_token()
+
+    assert cr_token == 'cr-token-1'
+
+
+def test_retrieve_cr_token_success_fail():
+    token_manager = ContainerTokenManager(
+        cr_token_filename='bogus-cr-token-file',
+    )
+
+    with pytest.raises(Exception) as err:
+        token_manager.retrieve_cr_token()
+
+    assert str(err.value) == 'Unable to retrieve the CR token value from file bogus-cr-token-file: [Errno 2] No such file or directory: \'bogus-cr-token-file\''
+
+
+@mock_iam_response
+def test_get_token_success():
+    token_manager = ContainerTokenManager(
+        cr_token_filename=cr_token_file,
+        iam_profile_name=MOCK_IAM_PROFILE_NAME,
+    )
+
+    cr_token = token_manager.access_token
```

Comment:
```suggestion
    access_token = token_manager.access_token
```
Using the name `cr_token` here is a bit confusing because you're actually assigning it from the "access_token" field which sort of implies that it's the IAM access token, and not the "cr token".

Reply:
Yep, you are totally right.